app/app.go

```diff
@@ -561,35 +559,43 @@ func NewStrideApp(
 	)
 	icaModule := ica.NewAppModule(&app.ICAControllerKeeper, &app.ICAHostKeeper)
 
-	// Create the middleware stacks
+	// Define the IBC stacks
+	// Stack one (ICAHost Stack) contains:
+	// - IBC
+	// - ICAHost
+	// - base app
+	icaHostIBCModule := icahost.NewIBCModule(app.ICAHostKeeper)
 
-	// Stack one contains
+	// Stack two (ICAController Stack) contains
 	// - IBC
-	// - ICA
-	// - icacallbacks
+	// - ICAController (middleware)
+	// - icacallbacks (middleware)
 	// - stakeibc
 	// - base app
-	var icamiddlewareStack porttypes.IBCModule
-	icamiddlewareStack = icacallbacksmodule.NewIBCModule(app.IcacallbacksKeeper, stakeibcIBCModule)
-	icamiddlewareStack = icacontroller.NewIBCModule(app.ICAControllerKeeper, icamiddlewareStack)
-	icaHostIBCModule := icahost.NewIBCModule(app.ICAHostKeeper)
+	icaCallbacksStack := icacallbacksmodule.NewIBCModule(app.IcacallbacksKeeper, stakeibcIBCModule)
+	icaControllerStack := icacontroller.NewIBCModule(app.ICAControllerKeeper, icaCallbacksStack)
 
-	// Stack two contains
+	// Stack three (Records Stack) contains
 	// - IBC
-	// - records
+	// - records (middleware)
 	// - transfer
 	// - base app
 	recordsStack := recordsmodule.NewIBCModule(app.RecordsKeeper, transferIBCModule)
 
 	// Create static IBC router, add transfer route, then set and seal it
+	// Note, authentication module packets are routed to the top level of the middleware stack
 	ibcRouter := ibcporttypes.NewRouter()
 	ibcRouter.
-		AddRoute(ibctransfertypes.ModuleName, recordsStack).
-		AddRoute(icacontrollertypes.SubModuleName, icamiddlewareStack).
+		// ICAHost Stack
 		AddRoute(icahosttypes.SubModuleName, icaHostIBCModule).
-		// Note, authentication module packets are routed to the top level of the middleware stack
```

Comment:
nit: leave this note in / update with a clarifying comment (I think it answers the question for the reviewer you had)

Reply:
I just moved it up to the line above `ibcRouter := ibcporttypes.NewRouter()`